File: src/test/testData/inspections/MissingRemoveCallbacksLeak.java
import android.app.Activity;
import android.os.Handler;

class MissingRemoveCallbacksLeak extends Activity {
    // Bad: Handler field without cleanup
    private Handler <warning descr="LeakLens: Handler 'mHandler' may cause a leak. Call removeCallbacks in onDestroy().">mHandler</warning> = new Handler();

    // Good: Handled in onDestroy
    private Handler mSafeHandler = new Handler();

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mSafeHandler.removeCallbacksAndMessages(null);
    }
}

File: src/test/testData/inspections/MissingRemoveCallbacksLeak.kt
import android.app.Activity
import android.os.Handler

class MissingRemoveCallbacksLeakKt : Activity() {
    // Bad: Handler field without cleanup
    private val <warning descr =
        "LeakLens: Handler 'mHandler' may cause a leak. Call removeCallbacks in onDestroy()." > mHandler < / warning > =
            Handler()

    // Good: Handled in onDestroy
    private val mSafeHandler = Handler()

    override fun onDestroy() {
        super.onDestroy()
        mSafeHandler.removeCallbacksAndMessages(null)
    }
}
